core/bubble_dragger.js

```diff
@@ -0,0 +1,282 @@
+/**
```

Comment:
The bubble dragger moves things that live on the bubble canvas instead of the block canvas.

Things on the bubble canvas are always in front of things on the block canvas.  This ensures that blocks don't look like they're inside bubbles, among other things.  Ask @rachel-fenichel if you want more details.

The bubble dragger has a lot of code that's similar to the block dragger.  I'd like to eventually combine them more, but structuring it this way makes it easier to experiment with bubble behaviour for things like clicks and drags.